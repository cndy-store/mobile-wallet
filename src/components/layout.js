import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    padding: 20
  },
  row: {
    flex: 1,
    alignSelf: 'stretch',
    justifyContent: 'center'
  }
});

class Container extends Component {
  render() {
    return <View style={styles.container} {...this.props} />;
  }
}

class Row extends Component {
  render() {
    return <View style={styles.row} {...this.props} />;
  }
}

export { Container, Row };
