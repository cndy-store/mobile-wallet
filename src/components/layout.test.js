import React from 'react';
import { StyleSheet } from 'react-native';
import { Container, Row } from './layout';

import renderer from 'react-test-renderer';

describe('Container', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds additional props', () => {
    const tree = renderer.create(<Container hello="world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Row', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Row />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds additional props', () => {
    const tree = renderer.create(<Row hello="world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
