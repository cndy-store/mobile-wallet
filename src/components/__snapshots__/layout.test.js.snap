// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container adds additional props 1`] = `
<View
  hello="world"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 20,
    }
  }
/>
`;

exports[`Container renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 20,
    }
  }
/>
`;

exports[`Row adds additional props 1`] = `
<View
  hello="world"
  style={
    Object {
      "alignSelf": "stretch",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;

exports[`Row renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;
